ci/deployment.yaml

```diff
@@ -44,12 +44,32 @@ spec:
              value: /deploy-config/deploy-config.json
            - name: HAIL_CI_OAUTH_TOKEN
              value: /secrets/oauth-token/oauth-token
+           - name: HAIL_CI_GITHUB_CONTEXT
 {% if deploy %}
-           - name: HAIL_WATCHED_BRANCHES
-             value: '[["hail-is/hail:main",true]]'
+             valueFrom:
+               secretKeyRef:
+                 name: ci-config
+                 key: github_context
 {% else %}
+             value: "ci-test"
+{% endif %}
            - name: HAIL_WATCHED_BRANCHES
-             value: '[["hail-ci-test/ci-test-{{create_ci_test_repo.token}}:master", true]]'
+{% if deploy %}
+             valueFrom:
+               secretKeyRef:
+                 name: ci-config
+                 key: watched_branches
+{% else %}
+             value: '[["hail-ci-test/ci-test-{{create_ci_test_repo.token}}:master", true, true]]'
+{% endif %}
+           - name: HAIL_CI_DEPLOY_STEPS
+{% if deploy %}
+             valueFrom:
+               secretKeyRef:
+                 name: ci-config
+                 key: deploy_steps
+{% else %}
+             value: '[]'
```

Comment:
It would probably be better actually to just have a build.yaml step that creates a `ci-config` in the non-default namespace that looks like:

```
github_context="ci-test"
watched_branches="[["hail-ci-test/ci-test-{{create_ci_test_repo.token}}:master", true, true]]"
deploy_steps="[]"
storage_uri="{{ global.test_storage_uri }}"
```

And then remove all of this logic in the deployment.yaml. Then the CI Makefile wouldn't need all of this repo token business.